### DISCUSSION TITLE
Trait for `Responder<'_,'_>` not satisfied

### DISCUSSION THREAD

**Discussion by aditya-mitra** [rwf2/Rocket#1819] [Questions] [answered]
I am very new to Rust and I encountered this error while try to build a simple web server using Rocket.

I have the following code:

```rs
use rocket_contrib::json::Json;
use serde::{Serialize,Deserialize};

#[derive(Serialize,Deserialize)]
struct NotFound {
    success: bool,
    message: String,
}

#[get("/_/admin", format = "json")]
pub fn index() -> Json<NotFound> {
    Json(NotFound {
        success: false,
        message: String::from("Not Found Custom 404"),
    })
}
```

On compilation, it gives me the following error:

**the trait bound `rocket_contrib::json::Json<routes::index_route::NotFound>: rocket::response::Responder<'_, '_>` is not satisfied**

![image](https://user-images.githubusercontent.com/55396651/129360113-dc7cf071-7572-4f2c-be1d-a00ebbfe7fd6.png)

The dependencies in my _cargo.toml_ are as follows:

```toml
[package]
name = "server"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
rocket = "0.5.0-rc.1"
serde = { version = "1.0", features = ["derive"] }
serde_json = "1.0"

[dependencies.rocket_contrib]
version = "*"
default-features = false
features = ["json"]
```

Can somebody please help me with what mistake I am making?

**Comment by jebrosen** [comment, marked as answer]
This looks like a duplicate of #1779 and a few others. Please see the CHANGELOG, especially regarding the removal of `rocket_contrib`: https://github.com/SergioBenitez/Rocket/blob/v0.5-rc/CHANGELOG.md#contrib-graduation.

(Incidentally, `version = "*"` is rarely a good choice for `Cargo.toml`. Among other issues, it allows `cargo update` to either update or downgrade that dependency on a whim, likely breaking your code that uses it.)

**Reply by aditya-mitra** [reply]
Thanks a lot @jebrosen 

I did not notice I had to use `features = ["json"]` and include the `serde` library as well in [documentation example](https://rocket.rs/v0.5-rc/guide/responses/#json)

Also thanks for the tip :1st_place_medal: 

I referred to many blogs and videos before posting this question. All the results which even appear on the top of search use `rocket_contrib`. They are outdated. I will write a blog post once I get a very good grasp of Rust. :smiley:

**Reply by JoshuaNitschke** [reply]
I found the documentation hard to follow as well, which is why I am here.  I think an example Cargo.toml file showing what needs to be set up to successfully configure JSON would be helpful in the documentation.
